ompi/mca/vprotocol/pessimist/vprotocol_pessimist_component.c

```diff
@@ -111,16 +111,22 @@ static int mca_vprotocol_pessimist_component_close(void)
 
 /** VPROTOCOL level functions (same as PML one)
   */
-static mca_vprotocol_base_module_t *mca_vprotocol_pessimist_component_init( int* priority,
-                                                                          bool enable_progress_threads,
-                                                                          bool enable_mpi_threads)
+static mca_vprotocol_base_module_t*
+mca_vprotocol_pessimist_component_init( int* priority,
+                                        bool enable_progress_threads,
+                                        bool enable_mpi_threads)
 {
     V_OUTPUT_VERBOSE(500, "vprotocol_pessimist: component_init");
     *priority = _priority;
 
     /* sanity check */
-    if(enable_mpi_threads)
-    {
+    if(enable_mpi_threads) {
+        /**
+         * Prevent the pessimistic protocol from activating if we are in a potentially multithreaded
+         * applications. The reason is that without tracking the thread initiator of messages it is
+         * extrmely difficult to provide a global message ordering in threaded-scenarios. Thus, the
+         * safe approach is to turn off the logger.
```

Comment:
Maybe we could have an override flag for people that really know what they are doing. It is fine to use MT as long as the matching and post-order remains deterministic per communicator (i.e. a communicator per thread approach would work fine).

Reply:
I have added such support via the ```vprotocol_pessimist_allow_thread_multiple``` MCA parameter.

Reply:
+1